TEMPLATE.md

```diff
@@ -0,0 +1,652 @@
+
+`sfentity`
+
+``` php
+/**
+ * @ORM\Entity()
+ * @ORM\Table(name="name")
+ */
+```
+
+`sfidcolumn`
+
+``` php
+/**
+ * @ORM\Column(name="id", type="integer", nullable=false)
+ * @ORM\Id()
+ * @ORM\GeneratedValue(strategy="IDENTITY")
+ */
+ private $id;
+```
+
+`sfstringcolumn`
+
+``` php
+/**
+ * @ORM\Column(name="", type="string", length=255)
+ */
+ private $;
+```
+
+`sfdecimalcolumn`
+
+``` php
+/**
+ * @ORM\Column(name="", type="decimal", scale=2)
+ */
+ private $;
+```
+
+`sftextcolumn`
+
+``` php
+/**
+ * @ORM\Column(name="", type="text")
+ */
+ private $;
+```
+
+`sfintegercolumn`
+
+``` php
+/**
+ * @ORM\Column(name="", type="integer")
+ */
+ private $;
+```
+
+`sfbooleancolumn`
+
+``` php
+/**
+ * @ORM\Column(name="", type="boolean")
+ */
+ private $;
+```
+
+`sfsmallintcolumn`
+
+``` php
+/**
+ * @ORM\Column(name="", type="smallint")
+ */
+ private $;
+```
+
+`sfbigintcolumn`
+
+``` php
+/**
+ * @ORM\Column(name="", type="bigint")
+ */
+ private $;
+```
+
+`sfdatetimecolumn`
+
+``` php
+/**
+ * @ORM\Column(name="", type="datetime")
+ */
+ private $;
+```
+
+`sfdatecolumn`
+
+``` php
+/**
+ * @ORM\Column(name="", type="date")
+ */
+ private $;
+```
+
+`sftimecolumn`
+
+``` php
+/**
+ * @ORM\Column(name="", type="time")
+ */
+ private $;
+```
+
+`sffloatcolumn`
+
+``` php
+/**
+ * @ORM\Column(name="", type="float")
+ */
+ private $;
+```
+
+`sfarraycolumn`
+
+``` php
+/**
+ * @ORM\Column(name="", type="array")
+ */
+ private $;
+```
+
+`sfobjectcolumn`
+
+``` php
+/**
+ * @ORM\Column(name="", type="object")
+ */
+ private $;
+```
+
+### Forms ###
+
+`sfform`
+
+``` php
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\OptionsResolver\OptionsResolverInterface;
+
+class NameType extends AbstractType
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setDefaultOptions(OptionsResolverInterface $resolver)
+    {
+        $resolver->setDefaults(array(
+            'data_class' => '',
+        ));
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getName()
+    {
+        return 'name';
+    }
+}
+```
+
+`sfdatatransformer`
+
+``` php
+use Symfony\Component\Form\DataTransformerInterface;
+use Symfony\Component\Form\Exception\UnexpectedTypeException;
+use Symfony\Component\Form\Exception\TransformationFailedException;
+
+class NameTransformer implements DataTransformerInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function transform($value)
+    {
+
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function reverseTransform($value)
+    {
+
+    }
+}
+```
+
+### Twig ###
+
+`sftwigextension`
```

Comment:
actually, this was missing from the actual XML file - it only appeared in the README. I'm not sure what other ones were also missing. But, I won't include this, because the PhpStorm plugin for Symfony includes this as a file template